common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -7,74 +7,83 @@ SPDX-License-Identifier: Apache-2.0
+	"errors"
 	"io/ioutil"
 	"os"
 	"testing"
 
-	"github.com/hyperledger/fabric/common/ledger/blkstorage"
 	"github.com/hyperledger/fabric/common/ledger/blockledger"
+	"github.com/hyperledger/fabric/common/ledger/blockledger/fileledger/mock"
 	"github.com/hyperledger/fabric/common/metrics/disabled"
 	"github.com/stretchr/testify/require"
 )
 
-type mockBlockStoreProvider struct {
-	blockstore *blkstorage.BlockStore
-	list       []string
-	error      error
-}
-
-func (mbsp *mockBlockStoreProvider) Open(ledgerid string) (*blkstorage.BlockStore, error) {
-	return mbsp.blockstore, mbsp.error
-}
-
-func (mbsp *mockBlockStoreProvider) List() ([]string, error) {
-	return mbsp.list, mbsp.error
-}
+func TestBlockStoreProviderErrors(t *testing.T) {
+	mockBlockStore := &mock.BlockStoreProvider{}
+	f := &fileLedgerFactory{
+		blkstorageProvider: mockBlockStore,
+		ledgers:            map[string]blockledger.ReadWriter{},
+	}
 
-func (mbsp *mockBlockStoreProvider) Close() {
-}
+	t.Run("list", func(t *testing.T) {
+		mockBlockStore.ListReturns(nil, errors.New("boogie"))
+		require.PanicsWithValue(
+			t,
+			"boogie",
+			func() { f.ChannelIDs() },
+			"Expected ChannelIDs to panic if storage provider cannot list channel IDs",
+		)
+	})
 
-func TestBlockstoreProviderError(t *testing.T) {
-	flf := &fileLedgerFactory{
-		blkstorageProvider: &mockBlockStoreProvider{error: fmt.Errorf("blockstorage provider error")},
-		ledgers:            make(map[string]blockledger.ReadWriter),
-	}
-	require.Panics(
-		t,
-		func() { flf.ChannelIDs() },
-		"Expected ChannelIDs to panic if storage provider cannot list channel IDs")
+	t.Run("open", func(t *testing.T) {
+		mockBlockStore.OpenReturns(nil, errors.New("woogie"))
+		_, err := f.GetOrCreate("foo")
+		require.EqualError(t, err, "woogie")
+		require.Empty(t, f.ledgers, "Expected no new ledger is created")
+	})
 
-	_, err := flf.GetOrCreate("foo")
-	require.Error(t, err, "Expected GetOrCreate to return error if blockstorage provider cannot open")
-	require.Empty(t, flf.ledgers, "Expected no new ledger is created")
+	t.Run("remove", func(t *testing.T) {
+		mockBlockStore.DropReturns(errors.New("oogie"))
+		err := f.Remove("foo")
+		require.EqualError(t, err, "oogie")
+	})
 }
 
 func TestMultiReinitialization(t *testing.T) {
 	metricsProvider := &disabled.Provider{}
 
-	dir, err := ioutil.TempDir("", "hyperledger_fabric")
+	dir, err := ioutil.TempDir("", "fileledger")
 	require.NoError(t, err, "Error creating temp dir: %s", err)
 	defer os.RemoveAll(dir)
 
-	flf, err := New(dir, metricsProvider)
+	f, err := New(dir, metricsProvider)
 	require.NoError(t, err)
-	_, err = flf.GetOrCreate("testchannelid")
+	_, err = f.GetOrCreate("testchannelid")
 	require.NoError(t, err, "Error GetOrCreate channel")
-	require.Equal(t, 1, len(flf.ChannelIDs()), "Expected 1 channel")
-	flf.Close()
+	require.Equal(t, 1, len(f.ChannelIDs()), "Expected 1 channel")
+	f.Close()
 
-	flf, err = New(dir, metricsProvider)
+	f, err = New(dir, metricsProvider)
```

Comment:
What happens if `New` is called twice without `Close` in the middle?

Reply:
We get a panic:

panic: Error opening leveldb: resource temporarily unavailable